sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormat.scala

```diff
@@ -65,6 +65,18 @@ trait FileFormat {
   }
 
   /**
+   * Returns concrete column vector class names for each column to be used in a columnar batch
+   * if this format supports returning columnar batch.
+   */
+  def vectorTypes(
+      sparkSession: SparkSession,
```

Comment:
I see.
To change the type may depend on each overriding function. Can we pass `sparkSession` as Option?
If `dataSchema` is not used now, is it make simple to drop `dataSchema`?